Port to Meson

wjt opened the pull request:
Everybody loves Meson.

Debian packaging half on https://github.com/endlessm/eos-installer/pull/104.

https://phabricator.endlessm.com/T33912

wjt committed:
Update libglnx snapshot

In particular, this snapshot has a meson.build file.

wjt committed:
Move libglnx submodule to subprojects

This is the path Meson likes.

wjt committed:
Port from intltools to xgettext

https://phabricator.endlessm.com/T33912

wjt committed:
Add Meson build system

Started on a Eurostar in February 2020, finished on a Eurostar in
October 2022.

https://phabricator.endlessm.com/T33912

wjt committed:
Remove Autotools build system

https://phabricator.endlessm.com/T33912

wjt committed:
Remove AUTHORS

This information comes from gnome-initial-setup, from which
eos-installer is derived. Most of the gnome-initial-setup code was
removed from this repo 4+ years ago, e.g. in
01b61c9ef7795308198e82bd1dc63a588df0dcb5 "Remove unused
gnome-initial-setup files". AUTHORS was similarly removed in
gnome-initial-setup 2 years ago.

The Git history is the canonical record of contributors.

https://phabricator.endlessm.com/T33912

wjt commented:
retest this please

wjt commented:
The build time on Jenkins is ~1m41s, down from ~2m45s.

~~TODO: libglnx spews `-Wpedantic` warnings. Squash these.~~

wjt committed:
Update libglnx (for reduced warning_level)

The referenced merge request and about a half-dozen others were merged.

https://phabricator.endlessm.com/T33912

dbnicholson commented on gnome-initial-setup/meson.build:
```diff
@@ -0,0 +1,30 @@
+gis_assistant_resources = gnome.compile_resources(
+    'gis-assistant-resources',
+    files('gis-assistant.gresource.xml'),
+)
```
Just a guess, but you'd need to use `export: true` in order to use this in the `gnome-image-installer` executable. Otherwise it can't resolve the symbols outside the `libgisutil` translation units.

https://mesonbuild.com/Gnome-module.html#gnomecompile_resources

wjt replied:
No dice, but I found a different workaround. I think it's because no symbol from `gis-assistant-resources.c` is explicitly referenced elsewhere (only implicitly) so it is dropped when linking the static library. `export: true` doesn't seem to make any difference in that regard. But when I add an explicit reference in `gis_assistant_class_init` all works fine. I don't understand linkers.

wjt replied:
Added that patch to the series.

dbnicholson commented on eos-installer-data/meson.build:
```diff
@@ -0,0 +1,35 @@
+install_data(
+    icon_dir / 'com.endlessm.Installer.png',
+    install_dir: datadir / icon_dir,
+)
```
No newline at end of file.

wjt replied:
Guess which one I wrote with GNOME Builder (defaults to no trailing newline) and which with Vim!

`.editorconfig` added.

dbnicholson replied:
Why would builder default to no trailing newline? :roll_eyes:

dbnicholson commented on gnome-image-installer/pages/confirm/meson.build:
```diff
@@ -0,0 +1,25 @@
+    link_with: libgisconfirm,
+    include_directories: include_directories('.'),
+)
+
```
Empty last line.

dbnicholson commented on gnome-image-installer/pages/diskimage/meson.build:
```diff
@@ -0,0 +1,26 @@
+    link_with: libgisdiskimage,
+    include_directories: include_directories('.'),
+)
+
```
Empty last line.

dbnicholson commented on gnome-image-installer/pages/disktarget/meson.build:
```diff
@@ -0,0 +1,25 @@
+    link_with: libgisdisktarget,
+    include_directories: include_directories('.'),
+)
+
```
Empty last line.

dbnicholson commented on gnome-image-installer/pages/finished/meson.build:
```diff
@@ -0,0 +1,25 @@
+    link_with: libgisfinished,
+    include_directories: include_directories('.'),
+)
+
```
Empty last line.

dbnicholson commented on gnome-image-installer/pages/install/meson.build:
```diff
@@ -0,0 +1,28 @@
+    link_with: libgisinstall,
+    include_directories: include_directories('.'),
+)
+
```
Empty last line.

dbnicholson reviewed (approved):
That all seems reasonable, although I'm a meson novice. I left some formatting nitpicks but otherwise it seems fine.

wjt committed:
assistant: Explicitly reference resources

Without this, it appears that gis-assistant-resources.o is dropped at
some point during the build process.

https://phabricator.endlessm.com/T33912

wjt committed:
Add .editorconfig

https://phabricator.endlessm.com/T33912

dbnicholson commented on .editorconfig:
```diff
@@ -0,0 +1,5 @@
+
+[*]
+end_of_line = lf
+insert_final_newline = true
```
This file itself has no trailing newline!

dbnicholson replied:
Since you're almost certainly asleep, I'll just clean it up and merge this.
